src/Compilers/CSharp/Portable/Lowering/SpillSequenceSpiller.cs

```diff
@@ -576,7 +583,11 @@ private BoundStatement UpdateStatement(BoundSpillSequenceBuilder builder, BoundS
             {
                 var lastSpillNode = (BoundSpillSequenceBuilder)newList[lastSpill];
                 builder.Include(lastSpillNode);
-                result.Add(lastSpillNode.Value);
+
+                var value = lastSpill > 0
+                    ? new Spiller(this).VisitExpression(ref builder, lastSpillNode.Value)
```

Comment:
>Would simply adjusting Spill helper to properly handling spilling of BoundLoweredIsPatternExpression instead work?

The `Spill` helper won't be called on this node, because the Value of the SpillSequence is not spilled unless it's not the last one in VisitExpressionList.

The requirement for spilling this node is kind of backwards. Anywhere else we spill all the nodes up to a SpillSequence, but we need to do the same for BoundLoweredIsPatternExpression only if there's (any) expressions preceding it.

Consider:
```cs
M(t + (e is <complex-pattern>));
```
When jumps are emitted the stack is empty, but once we're visiting BoundLoweredIsPatternExpression that's no longer the case and MarkLabel fails.